docs/reference_guides/model_libraries/generic/costing/process_costing.rst

```diff
@@ -0,0 +1,865 @@
+Base cost                   :math:`base\_cost`             unitless     Base cost (base cost per unit * number of units)
+Number of units             :math:`number\_of\_units`      unitless     Number of units to be costed (to take advantage of the economics of scale)
+=========================== ============================== ============ ===============================================================================
+
+.. note:: number of units by default is fixed to 1 and the user must unfix this variable to optimize the number of units. Also, `number of units` can be built as a continuous variable or an integer variable. If the latter, the user must provide an mip solver. Use the global costing argument for this purpose (integer_n_units=True or False).
+
+Example
+^^^^^^^
+Below is a simple example of how to add cost correlations to a flowsheet including a heat exchanger using the default IDAES costing module.
+
+
+.. code:: python
+
+    from pyomo.environ import ConcreteModel, SolverFactory
+    from pyomo.util.calc_var_value import calculate_variable_from_constraint
+    from idaes.core import FlowsheetBlock
+    from idaes.models.unit_models.heat_exchanger import HeatExchanger, HeatExchangerFlowPattern
+    from idaes.models.properties import iapws95
+    from idaes.core.util.model_statistics import degrees_of_freedom
+
+    from idaes.models.costing.SSLW import SSLWCosting, SSLWCostingData
+    from idaes.core import UnitModelCostingBlock
+    from idaes.models.costing.SSLW import HeaterMaterial, HeaterSource
+    
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    
+    m.fs.properties = iapws95.Iapws95ParameterBlock()
+    
+    m.fs.unit = HeatExchanger(default={
+                "shell": {"property_package": m.fs.properties},
+                "tube": {"property_package": m.fs.properties},
+                "flow_pattern": HeatExchangerFlowPattern.countercurrent})
+    # set inputs
+    m.fs.unit.shell_inlet.flow_mol[0].fix(100)     # mol/s
+    m.fs.unit.shell_inlet.enth_mol[0].fix(3500)    # j/s
+    m.fs.unit.shell_inlet.pressure[0].fix(101325)  # Pa 
+    
+    m.fs.unit.tube_inlet.flow_mol[0].fix(100)
+    m.fs.unit.tube_inlet.enth_mol[0].fix(4000)
+    m.fs.unit.tube_inlet.pressure[0].fix(101325.0)
+    
+    m.fs.unit.area.fix(1000)  # m2
+    m.fs.unit.overall_heat_transfer_coefficient.fix(100)  # W/m2K
+    
+    m.fs.costing = SSLWCosting()
+    m.fs.H101.costing = UnitModelCostingBlock(
+        default={
+            "flowsheet_costing_block": m.fs.costing,
+            "costing_method": SSLWCostingData.cost_fired_heater,
+            "costing_method_arguments": {
+                    "material_type": HeaterMaterial.CarbonSteel,
+                    "heat_source": HeaterSource.Fuel
+            }
+        }
+    )
+
+    m.fs.unit.initialize()
+
+    opt = SolverFactory('ipopt')
+    opt.options = {'tol': 1e-6, 'max_iter': 50}
+    results = opt.solve(m, tee=True)
+
+Units
```

Comment:
Units of Measurement - you need to be clear here to avoid confusion with "unit model" and "unit operation".

Secondly, I would leave most of the details out - just note that unit conversion are handled automatically. Most users will have no idea about the metadata dictionary.

Do remind users of the naming for the currency units however (this should also have been mentioned in the general discussion, but it will help to repeat ourselves on this).